engine/access/ingestion/collection_syncer.go

```diff
@@ -0,0 +1,389 @@
+// all collections associated with finalized blocks starting from the
+// last fully synced height. It works by periodically scanning the finalized
+// block range, identifying missing collections, and triggering requests
+// to fetch them from the network. Once collections are retrieved, it
+// ensures they are persisted in the local collection and transaction stores.
+//
+// The syncer maintains a persistent, strictly monotonic counter
+// (`lastFullBlockHeight`) to track the highest finalized block for which
+// all collections have been fully indexed. It uses this information to
+// avoid redundant processing and to measure catch-up progress.
+//
+// It is meant to operate in a background goroutine as part of the
+// node's ingestion pipeline.
+type CollectionSyncer struct {
+	logger                   zerolog.Logger
+	collectionExecutedMetric module.CollectionExecutedMetric
+
+	state     protocol.State
+	requester module.Requester
+
+	blocks       storage.Blocks
+	collections  storage.Collections
+	transactions storage.Transactions
+
+	//TODO: is full block a common term in this bounded context?
+	// maybe rename to lastSyncedBlockHeight? in this context it looks reasonable
+	lastFullBlockHeight *counters.PersistentStrictMonotonicCounter
+}
+
+// NewCollectionSyncer creates a new CollectionSyncer responsible for downloading,
+// tracking, and indexing missing collections.
+func NewCollectionSyncer(
+	logger zerolog.Logger,
+	collectionExecutedMetric module.CollectionExecutedMetric,
+	requester module.Requester,
+	state protocol.State,
+	blocks storage.Blocks,
+	collections storage.Collections,
+	transactions storage.Transactions,
+	lastFullBlockHeight *counters.PersistentStrictMonotonicCounter,
+) *CollectionSyncer {
+	collectionExecutedMetric.UpdateLastFullBlockHeight(lastFullBlockHeight.Value())
+
+	return &CollectionSyncer{
+		logger:                   logger,
+		state:                    state,
+		requester:                requester,
+		blocks:                   blocks,
+		collections:              collections,
+		transactions:             transactions,
+		lastFullBlockHeight:      lastFullBlockHeight,
+		collectionExecutedMetric: collectionExecutedMetric,
+	}
+}
+
+// RequestCollections continuously monitors and triggers collection sync operations.
+// It handles on startup collection catchup, periodic missing collection requests, and full block height updates.
+func (s *CollectionSyncer) RequestCollections(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+	requestCtx, cancel := context.WithTimeout(ctx, defaultCollectionCatchupTimeout)
+	defer cancel()
+
+	// on start-up, AN wants to download all missing collections to serve it to end users
+	err := s.downloadMissingCollections(requestCtx)
+	if err != nil {
+		s.logger.Error().Err(err).Msg("error downloading missing collections")
+	}
+	ready()
+
+	requestCollectionsInterval := time.NewTicker(defaultMissingCollsRequestInterval)
```

Comment:
we typically use "*Interval" for the actual time.Duration value. how about `requestCollectionsTicker` or `requestCollections`?